src/xo-app/backup/restore/index.js

```diff
@@ -0,0 +1,215 @@
+import find from 'lodash/find'
+import forEach from 'lodash/forEach'
+import Icon from 'icon'
+import Link from 'react-router/lib/Link'
+import map from 'lodash/map'
+import orderBy from 'lodash/orderBy'
+import React, { Component } from 'react'
+import size from 'lodash/size'
+import { connectStore, formatSize } from 'utils'
+import { createGetObject, createGetObjectsOfType } from 'selectors'
+import { DropdownButton, MenuItem } from 'react-bootstrap-4/lib'
+import { info, error } from 'notification'
+import { FormattedDate } from 'react-intl'
+import { Row, Col } from 'grid'
+
+import {
+  importBackup,
+  importDeltaBackup,
+  listRemote,
+  subscribeRemotes
+} from 'xo'
+
+const deltaBuilder = (backups, uuid, name, tag, value) => {
+  let deltaBackup = backups[uuid]
+    ? backups[uuid]
+    : backups[uuid] = {}
+
+  deltaBackup = deltaBackup[name]
+    ? deltaBackup[name]
+    : deltaBackup[name] = {}
+
+  deltaBackup = deltaBackup[tag]
+    ? deltaBackup[tag]
+    : deltaBackup[tag] = []
+
+  deltaBackup.push(value)
+}
+
+const isEmptyRemote = remote => !remote.backups || !(size(remote.backups.delta) || size(remote.backups.other))
+
+@connectStore(() => {
+  const object = createGetObject((_, props) => props.id)
+  return (state, props) => ({object: object(state, props)})
+})
+class ObjectName extends Component {
+  render () {
+    const { object } = this.props
+    return <span>{object && object.name_label}</span>
+  }
+}
+
+@connectStore(() => {
+  const writableSrs = createGetObjectsOfType('SR').filter(
+    [ sr => sr.content_type !== 'iso' ]
+  ).sort()
+  return (state, props) => {
+    return { writableSrs: writableSrs(state, props) }
+  }
+})
+export default class Restore extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      remotes: []
+    }
+  }
+
+  componentWillMount () {
+    this.componentWillUnmount = subscribeRemotes(rawRemotes => {
+      const { remotes } = this.state
+      forEach(remotes, r => {
+        if (r.backups) {
+          const freshRemote = find(rawRemotes, {id: r.id})
+          freshRemote && (freshRemote.backups = r.backups)
+        }
+      })
+      this.setState({
+        remotes: orderBy(rawRemotes, ['name'])
+      })
+    })
+  }
+
+  _list = async id => {
+    try {
+      const files = await listRemote(id)
+      const { remotes } = this.state
+      const remote = find(remotes, {id})
+      if (remote) {
+        const backups = remote.backups = {
+          delta: {},
+          other: []
+        }
+
+        forEach(files, file => {
+          const arr = /^vm_delta_(.*)_([^\/]+)\/([^_]+)_(.*)$/.exec(file)
+
+          if (arr) {
+            const [ , tag, uuid, date, name ] = arr
+            const value = {
+              path: file,
+              date
+            }
+            deltaBuilder(backups.delta, uuid, name, tag, value)
+          } else {
+            backups.other.push(file)
+          }
+        })
+      }
```

Comment:
Yes and no, the file hierarchy on the server side should not be exposed to the UIs.

If this is too much work, we'll think about it later.